### PR TITLE
TST: fixturize, collect

### PR DESCRIPTION
**Pull request by jbrockmendel**
- [ ] closes #xxxx
- [ ] tests added / passed
- [ ] Ensure all linting tests pass, see [here](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit) for how to run them
- [ ] whatsnew entry

### CODE REVIEW

File: pandas/conftest.py [hunk 1]
```diff
@@ -921,6 +921,23 @@ def all_binary_operators(request):
     return request.param
 
 
+@pytest.fixture(
```
Comment (jreback) [inline pandas/conftest.py:924]: isn't this all_compare_operators( ?

**Comment by jbrockmendel** [comment]
That’s the dunder strings

File: pandas/conftest.py (hunk 1 above)
Comment (jreback) [reply pandas/conftest.py:924]: we don't really distinguish the naming here.  e.g. for arithmetic we have
`all_binary_operations` (using operator) and `all_arithmetic_operators`. 

so ideally have some good pattern here. 

maybe 
`all_comparison_operators` (use `operator`)
`all_comparison_dunder` (use `dunder`)

would have to fix a bunch of things but prob worth it.

Comment (jbrockmendel) [reply pandas/conftest.py:924]: yah not something i want to futz with

Comment (jreback) [reply pandas/conftest.py:924]: sure but we cannot add it like this, so need to do something. e.g. move next to the others & rename this one to conform.

Comment (jbrockmendel) [reply pandas/conftest.py:924]: will move this down adjacent to all_compare_operators

Comment (jbrockmendel) [reply pandas/conftest.py:924]: looks like most of the places here we use the dunder one then do a `getattr(operator, all_comparison_operators.strip("__"))`, so will just replace those usages with this

File: pandas/tests/extension/base/ops.py [hunk 2]
```diff
@@ -154,23 +153,22 @@ def _compare_other(self, ser: pd.Series, data, op_name: str, other):
...
+    def test_compare_array(self, data, comparison_op):
         ser = pd.Series(data)
         other = pd.Series([data[0]] * len(data))
-        self._compare_other(ser, data, op_name, other)
+        self._compare_other(ser, data, comparison_op, other)
 
-    @pytest.mark.parametrize("box", [pd.Series, pd.DataFrame])
-    def test_direct_arith_with_ndframe_returns_not_implemented(self, data, box):
+    def test_direct_arith_with_ndframe_returns_not_implemented(
+        self, data, frame_or_series
```
Comment (jorisvandenbossche) [inline pandas/tests/extension/base/ops.py:166]: Those base extension tests are used by downstream projects, so we should be a bit more careful with changing them (at least when for "clean up" purposes). Or use fixtures defined in `tests/extension/conftest.py` as you did in https://github.com/pandas-dev/pandas/pull/44138. 
I have changed back this specific line in https://github.com/pandas-dev/pandas/pull/44332

**Pull request merged**